Update arch diagram to match terminology

thibmeu opened the pull request:
Update Figure 1 to include components from the Terminology.

* Credentials is owned by an Issuer
* Collection of receipt is done by a Relying Party
* Make of Transparency Statement is done by an Auditor, which is a special Relying Party

The newly generated diagram looks as follow
![architecture difference](https://github.com/user-attachments/assets/aa521d21-eb29-420f-982b-63059b2cf686)

SteveLasker commented on draft-ietf-scitt-architecture.md:
```diff
@@ -374,18 +375,24 @@ The arrows indicate the flow of information.
-     .--------+---------.      .----------+-----.
-    / Collect Receipts /      /   Replay Log   /
-   '------------------'      '----------------'
+              V                           |   |
```
```suggestion
              |                           |   |
              V                           |   |
```

SteveLasker commented on draft-ietf-scitt-architecture.md:
```diff
@@ -347,9 +347,9 @@ The arrows indicate the flow of information.
-       v                 .-------------.
-  .----+----.           |  Credentials  |
- | Statement |           '----+--+-----'
+       v            +----------------------+
+  .----+----.       |  Issuer Credentials  |
+ | Statement |      +---------+--+---------+
```
```suggestion
       v             +--------------------+
  .----+----.        | Issuer Credentials |
 | Statement |       +--------+--+--------+
```

SteveLasker commented on draft-ietf-scitt-architecture.md:
```diff
@@ -381,11 +381,15 @@ The arrows indicate the flow of information.
               |   / Verify Transparent /  |
               |  /      Statement     /   |
-              | '--------------------'    |
+              | '-----------+--------'|   |
+              |             | Auditor |   |
+              |             +---------+   |
```
```suggestion
              | '----------+---------+    |
              |            | Auditor |    |
              |            +---------+    |
```

SteveLasker commented on draft-ietf-scitt-architecture.md:
```diff
@@ -381,11 +381,15 @@ The arrows indicate the flow of information.
-     .--------+---------.      .----------+-----.
-    / Collect Receipts /      /   Replay Log   /
-   '------------------'      '----------------'
+  .-----------+----------.       .----------+-----.
+ /  Collecting Receipt  /       /   Replay Log   /
+'-------+--------------'|      '-+--------------'|
+        | Relying Party |        | Relying Party |
+        +---------------+        +---------------+
```
```suggestion
  .-----------+----------.    .-----------+------.
 /  Collecting Receipt  /    /    Replay Log    /
'------+---------------+    '--+---------------+
       | Relying Party |       | Relying Party |
       +---------------+       +---------------+
```

SteveLasker reviewed (approved):
Thanks, @thibmeu 
LGTM

Minor tweaks to fix the aasvg corners:
![image](https://github.com/user-attachments/assets/3252563d-78e3-4d32-bb94-90c39699db23)

JAG-UK reviewed (approved):
Lgtm

yogeshbdeshpande commented:
@thibmeu I have a slight concern with the diagram. Normally Replay log is explicitly done by Auditor so Under Replay Log, if the role is Auditor that is a more correct representation than a Generic Relying Party what you have currently!

yogeshbdeshpande commented:
Also I have a concern, Verify Transparent Statements is a job of any Relying Party, not specifically for the Auditor?

Did you intend to attach, the Auditor role to Verify Transparent Statements?

yogeshbdeshpande reviewed (changes requested):
Kindly answer these questions please before I can approve, happy to discuss face to face!

thibmeu commented:
> Also I have a concern, Verify Transparent Statements is a job of any Relying Party, not specifically for the Auditor?

From the terminology
```
Auditor: checks the correctness and consistency of all Transparent Statements issued by a Transparency Service
Relying Party: consumes Transparent Statements, verifying their proofs and inspecting the Statement payload
```

I read this as the Auditor builds verified transparent statement, while the replay log is a specialised relying Party but not necessarily an auditor

Removing Auditor is also a possibility to not put too many entities, since it's relying parties anyway. What do you think?

thibmeu commented:
@yogeshbdeshpande I've addressed your comments following a discusion we had in person
here is the updated diagram

![Screenshot from 2024-11-04 11-41-47](https://github.com/user-attachments/assets/ca0837ae-7b15-48da-9e56-9aea72e07078)

yogeshbdeshpande reviewed (approved):
Thank you for making this important change!
Looks Good to Me
